upstairs/src/lib.rs

```diff
@@ -8789,22 +8788,26 @@ impl GuestWork {
          * A gw_id that already finished and results were sent back to
          * the guest could still have an outstanding ds_id.
          */
-        if let Some(mut gtos_job) = self.active.remove(&gw_id) {
+        let mut gtos_job_done = false;
+        if let Some(gtos_job) = self.active.get_mut(&gw_id) {
```

Comment:
You could avoid mutation with something like
```rust
        let gtos_job_done = if let Some(gtos_job) = self.active.get_mut(&gw_id) {
            // ...
            gtos_job.submitted.is_empty()
        } else {
            // ...
            false
        };
```

Reply:
nice! 8fa5a530c3fc2389ac51ff7e985c44afb5995af7